hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneClientAdapterImpl.java

```diff
@@ -784,6 +804,28 @@ private List<FileStatusAdapter> listStatusVolume(String volumeStr,
     return res;
   }
 
+  /**
+   * Helper for OFS listStatus on a bucket to get all snapshots.
+   */
+  private List<FileStatusAdapter> listStatusBucketSnapshot(
+      String volumeName, String bucketName, URI uri) throws IOException {
+    List<OzoneSnapshot> snapshotList =
+        objectStore.listSnapshot(volumeName, bucketName);
```

Comment:
I assume this `listSnapshot` by default filters out all snapshot that are not `ACTIVE` ? For example some snapshots are deleted pending reclaimation would still remain in the OM table.

Reply:
I didn't create `listSnapshot()`, [here](https://github.com/xBis7/ozone/blob/HDDS-7513/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java#L1200-L1206) is the implementation. There are other places or commands using this method. If we add the filter below here, we wouldn't be that consistent but on the other hand we might need `listSnapshot()` to return all the snapshots and not just the `ACTIVE` ones.

Reply:
Agreed. We should do this in another JIRA.